nitime/timeseries.py

```diff
@@ -309,7 +316,11 @@ def mean(self, *args, **kwargs):
         return ret
 
     def ptp(self, *args, **kwargs):
-        ret = TimeArray(np.ndarray.ptp(self, *args, **kwargs),
+        if _NP_2:
+            ptp = np.ptp
+        else:
+            ptp = np.ndarray.ptp
+        ret = TimeArray(ptp(self, *args, **kwargs),
```

Comment:
New suggestion:

```suggestion
        ret = TimeArray(np.ptp(self.view(np.ndarray), *args, **kwargs),
```

Works on my numpy 1.x tests. Seems like it should work with numpy 2 as well.

Reply:
Oh, tests are passing. Hmm.

Reply:
Yeah I think we're okay actually with your original code. Not sure how I got the recursion problem when I tried before